mcafee-agent/elx/init.sls

```diff
@@ -67,6 +67,8 @@ Install McAfee Agent:
     - cwd: '/root'
     - require:
       - file: Stage McAfee Install Archive
+    - success_retcodes:
+      - 1
```

Comment:
Isn't this supposed to be a `2`?
```suggestion
      - 2
```

Reply:
It's `1` in this case.

`[ERROR   ][2625] {'pid': 2750, 'retcode': 1, 'stdout': 'An higher or same version of the agent is already installed. Installation cannot continue', 'stderr': ''}`

Reply:
Actually, I think we're trying to handle that scenario in a limited way here, skipping the install if already installed:

https://github.com/plus3it/mcafee-agent-formula/blob/e5ee0801f300a8aa2da645a2e5baf09f78458033/mcafee-agent/elx/init.sls#L70-L73

So it shouldn't be running the installer at all, I think?

Reply:
Okay.  Yeah, I don't see an option to check install status. I tried `-u` and it gives the same result.

```
]# sh ./install.sh -h
./install.sh: option requires an argument -- h
Usage: ./install.sh [-i|-u|-b|-p|-r|-R|-h  ]
-i : fresh install
-u : upgrade install
-b :Upgrade Agent only , no server info will be updated
-p : generate provisioning rpm/deb
-r : Use this option while installing/upgrading, to run macmnsvc as root user
-R : Set Relay Server ('IP1:Port1[;IP2:Port2;...]')
-h : show this help
```

Until your patch is incorporated, maybe I'll try to see if I can add a state to check for an existing install.

Reply:
> Actually, I think we're trying to handle that scenario in a limited way here, skipping the install if already installed:
> 
> https://github.com/plus3it/mcafee-agent-formula/blob/e5ee0801f300a8aa2da645a2e5baf09f78458033/mcafee-agent/elx/init.sls#L70-L73
> 
> So it shouldn't be running the installer at all, I think?

Right.  We're not even providing an rpm.

Reply:
Oh.  I missed the default values in `map.jinja'.  Let me check what's on the system I'm using for testing.

Reply:
It seems something is not right with the handling of `unless`.  I'm hard coding the command and it's still not working.  The command is still being executed.
```
Install McAfee Agent:
  cmd.run:
    - name: 'sh /root/install.sh {{ mcafee.installer_opts }}'
    - cwd: '/root'
    - require:
      - file: Stage McAfee Install Archive
    - unless:
      - 'rpm --quiet -q MFEcma'
{%- for key_file in mcafee.key_files %}
      - 'test -s {{ mcafee.keystore_directory }}/{{ key_file }}'
{%- endfor %}
```